metadata-dao/src/main/java/com/linkedin/metadata/dao/equality/DefaultEqualityTester.java

```diff
@@ -1,21 +1,24 @@
 package com.linkedin.metadata.dao.equality;
 
+import com.linkedin.data.template.DataTemplate;
+import com.linkedin.data.template.DataTemplateUtil;
 import javax.annotation.Nonnull;
 
 /**
- * A {@link EqualityTester} that relies on the default implementation of {@link Object#equals(Object)}.
+ * A {@link EqualityTester} that uses {@link DataTemplateUtil#areEqual(DataTemplate, DataTemplate)} to check for
+ * semantic equality.
  */
-public class DefaultEqualityTester<T> implements EqualityTester<T> {
+public class DefaultEqualityTester<T extends DataTemplate> implements EqualityTester<T> {
 
   /**
    * Creates a new instance of {@link DefaultEqualityTester}.
    */
-  public static <CLASS> DefaultEqualityTester<CLASS> newInstance() {
-    return new DefaultEqualityTester<CLASS>();
+  public static <CLASS extends DataTemplate> DefaultEqualityTester<CLASS> newInstance() {
+    return new DefaultEqualityTester<>();
   }
 
   @Override
   public boolean equals(@Nonnull T o1, @Nonnull T o2) {
-    return o1.equals(o2);
+    return DataTemplateUtil.areEqual(o1, o2);
```

Comment:
So DataTemplateUtil.areEqual(o1, o2) will not ignore unrecognized fields - is that intended? https://github.com/linkedin/rest.li/blob/master/data/src/main/java/com/linkedin/data/template/DataTemplateUtil.java#L1036